include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+        {
+            std::cerr << "[SEQAN INFO] :: Thank you for using SeqAn!\n"
+                      << "[SEQAN INFO] :: You might want to regsiter you app for support and version check features?!\n"
+                      << "[SEQAN INFO] :: Just send us an email to seqan@team.fu-berlin.de with your app name and version number.\n"
+                      << "[SEQAN INFO] :: If you don't want to recieve this message anymore set --version_check OFF\n\n";
+            myfile.close();
+            return false;
+        }
+        else
+        {
+            myfile.close();
+            return false;
+        }
+    }
+    else
+    {
+        return false;
+    }
+    return true;
+}
+
+// ----------------------------------------------------------------------------
+// Function _isSmaller()
+// ----------------------------------------------------------------------------
+inline bool _isSmaller(seqan::String<int> & left, seqan::String<int> & right)
+{
+    for (unsigned i = 0; i < length(left); ++i)
+    {
+        if (left[i] < right[i])
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+// ----------------------------------------------------------------------------
+// Function _callServer()
+// ----------------------------------------------------------------------------
+inline bool _callServer(VersionCheck me, bool version_file_exists)
+{
+    // system call
+    // http response is stored in a file '.config/seqan/{app_name}_version'
+    if (system(me._command.c_str()))
+    {
+        std::string version_file = me._path + "/" + me._name + "_version.txt";
+        if (!version_file_exists)
+        {
+            std::ofstream version_file_out (version_file.c_str());
+            if (version_file_out.is_open())
+            {
+                version_file_out << "UNABLE TO CALL HOME.\n";
+                version_file_out.close();
+            }
+        }
+        else // just set last modified date to current and do not override possible version information
+        {
+#if defined(PLATFORM_WINDOWS)
+            FILETIME curr;
+            GetSystemTimeAsFileTime(&curr);
+            HANDLE hFile;
+            hFile = CreateFile(version_file.c_str(), GENERIC_WRITE, FILE_SHARE_READ,  NULL,  OPEN_EXISTING,  FILE_ATTRIBUTE_NORMAL, NULL);
+            SetFileTime(hFile, NULL, NULL, &curr);
+            CloseHandle(hFile);
+#else
+            time_t curr;
+            time(&curr); // get current time
+            struct utimbuf puttime;
+            puttime.modtime = curr;
+            puttime.actime = curr;
+            utime(version_file.c_str(), &puttime);
+#endif
+        }
+        return false;
+    }
+    return true;
+}
+
+// ----------------------------------------------------------------------------
+// Function checkForNewerVersion()
+// ----------------------------------------------------------------------------
+inline bool checkForNewerVersion(VersionCheck & me)
```

Comment:
I think this entire function should run in a separate thread.
And even if the operations are quite small. There is no guarantee, that the thread is terminated before the file is written too and closed properly.
So it would be better to wrap it in a class, whose destructor can make sure, that all file handlers are closed correctly.
In case it could also utilise a rollback functionality.